File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/view/adapters/CardsPagerAdapter.java
package com.dbottillo.mtgsearchfree.view.adapters;

import android.content.Context;
import android.support.v4.view.PagerAdapter;
import android.view.View;
import android.view.ViewGroup;

import com.dbottillo.mtgsearchfree.model.CardsCollection;
import com.dbottillo.mtgsearchfree.model.MTGCard;
import com.dbottillo.mtgsearchfree.view.views.MTGCardView;

import java.util.List;

public class CardsPagerAdapter extends PagerAdapter {

    private Context context;
    private boolean showImage;
    private CardsCollection cards;

    public CardsPagerAdapter(Context context, boolean showImage, CardsCollection cards) {
        this.context = context;
        this.showImage = showImage;
        this.cards = cards;
    }

    public int getCount() {
        return cards.getList().size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        MTGCardView view = new MTGCardView(context);
        view.load(cards.getList().get(position), showImage);
        container.addView(view);
        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object view) {
        container.removeView((View) view);
    }

    public CharSequence getPageTitle(int position) {
        MTGCard card = cards.getList().get(position);
        if (cards.isDeck()) {
            return card.getName() + " (" + card.getQuantity() + ")";
        }
        return card.getName();
    }

    public MTGCard getItem(int currentItem) {
        return cards.getList().get(currentItem);
    }

}

File: mtgSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/view/adapters/CardsPagerAdapter.kt
package com.dbottillo.mtgsearchfree.view.adapters

import android.content.Context
import android.support.v4.view.PagerAdapter
import android.view.View
import android.view.ViewGroup
import com.dbottillo.mtgsearchfree.resources.MTGCard
import com.dbottillo.mtgsearchfree.view.views.MTGCardView
import java.util.*

class CardsPagerAdapter(var context: Context, var deck: Boolean, var cards: ArrayList<MTGCard>) : PagerAdapter() {

    override fun getCount(): Int {
        return cards.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view == `object`
    }

    override fun instantiateItem(collection: ViewGroup, position: Int): Any {
        var view = MTGCardView(context)
        view.load(cards.get(position), true)
        collection.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any?) {
        container.removeView(`object` as View)
    }

    override fun getPageTitle(position: Int): CharSequence {
        val card = cards[position]
        if (deck) {
            return card.name + " (" + card.quantity + ")"
        }
        return card.name
    }
}
